### ISSUE TITLE
Implement MutationObservers

### ISSUE THREAD

**Issue by matthewp** [canjs/can-zone#38]
This is how one Promise library does it:

```js
function initMutationObserver(MutationObserver) {
		var scheduled;
		var node = document.createTextNode('');
		var o = new MutationObserver(run);
		o.observe(node, { characterData: true });

		function run() {
			var f = scheduled;
			scheduled = void 0;
			f();
		}

		var i = 0;
		return function (f) {
			scheduled = f;
			node.data = (i ^= 1);
		};
	}
```

**Comment by matthewp** [comment]
Important thing here is we want to only detect when MutationObserver is being used for this reason. If we say:

1. Observing an empty TextNode
2. That is not in the DOM

If that's safe enough to implement.

**Comment by matthewp** [comment]
Closed by #63
